server/src/test/java/com/objectcomputing/checkins/services/checkinnotes/CheckinNoteControllerTest.java

```diff
@@ -0,0 +1,272 @@
+
+    }
+
+    @Test
+    void testCreateNullCheckinNote(){
+        CheckinNote cNote = new CheckinNote(UUID.randomUUID(),UUID.randomUUID(), "test");
+
+        when(checkinNoteServices.save(eq(cNote))).thenReturn(cNote);
+   
+        final HttpRequest<String> request = HttpRequest.POST("", "");
+        HttpClientResponseException responseException = assertThrows(HttpClientResponseException.class,
+        () -> client.toBlocking().exchange(request, Map.class));
+   
+        JsonNode body = responseException.getResponse().getBody(JsonNode.class).orElse(null);
+        JsonNode errors = Objects.requireNonNull(body).get("_embedded").get("errors");
+        JsonNode href = Objects.requireNonNull(body).get("_links").get("self").get("href");
+        List<String> errorList = List.of(errors.get(0).get("message").asText(), errors.get(1).get("message").asText())
+                .stream().sorted().collect(Collectors.toList());
+
+        assertEquals("Required Body [checkinNote] not specified", errors.asText());
+        assertEquals(request.getPath(), href.asText());
+        assertEquals(HttpStatus.BAD_REQUEST, responseException.getStatus());
+     
+    }
+
+
+    @Test
+    void testDeleteCheckinNote(){
+        UUID uuid = UUID.randomUUID();
+
+        final HttpRequest<UUID> request = HttpRequest.DELETE(uuid.toString());
+        final HttpResponse<String> response = client.toBlocking().exchange(request, String.class);
+
+        assertEquals(HttpStatus.OK, response.getStatus());
+
+    }
+
+    @Test
+    void testReadAllCheckinNote(){
+      Set<CheckinNote> checkinNote = Set.of(
+        new CheckinNote(UUID.randomUUID(),UUID.randomUUID(), "test") ,
+        new CheckinNote(UUID.randomUUID(),UUID.randomUUID(), "test")    
+        );
+
+        when(checkinNoteServices.readAll()).thenReturn(checkinNote);
+        final HttpRequest<UUID> request = HttpRequest.GET("all");
+        final HttpResponse<Set<CheckinNote>> response = client.toBlocking().exchange(request, Argument.setOf(CheckinNote.class));
+
+        assertEquals(checkinNote, response.body());
+        assertEquals(HttpStatus.OK, response.getStatus());
+
+    }
+
+    @Test
+    void testReadCheckinNote(){
+        CheckinNote cNote = new CheckinNote(UUID.randomUUID(), UUID.randomUUID(), UUID.randomUUID(),"test");
+
+        when(checkinNoteServices.read(cNote.getId())).thenReturn(cNote);
+        final HttpRequest<UUID> request = HttpRequest.GET(String.format("/%s",cNote.getId().toString()));
+        final HttpResponse<CheckinNote> response = client.toBlocking().exchange(request, CheckinNote.class);
+
+        assertEquals(cNote, response.body());
+        assertEquals(HttpStatus.OK, response.getStatus());
+    }
+
+    @Test
+    void testReadCheckinNoteNotFound(){
+        CheckinNote cNote = new CheckinNote(UUID.randomUUID(), UUID.randomUUID(), UUID.randomUUID(),"test");
+
+        when(checkinNoteServices.read(cNote.getId())).thenReturn(cNote);
```

Comment:
Should return null not cNote. You are trying to mock that it didn't find the checkin note.